### ISSUE TITLE
Saving native MongoDB and Druid queries fails due to incomplete `results_metadata`

### ISSUE THREAD

**Issue by ronny35** [metabase/metabase#12265]
Hi,

When I perform a new native MongoDB query, question is not saved.

04-04 09:26:19 ERROR analyze.query-results :: Error generating insights for column: {:name updatedAt~~~default}
clojure.lang.ExceptionInfo: Output of stored-column-metadata->result-column-metadata does not match schema:

In main case, I do this test :
1. Create a new question (with no filter) => **Save is OK**.
2. Edit this question and go to "native query" => **Save fails**.

Metabase version: 0.35.1
Metabase hosting environment: Docker
Database used: MongoDB

**Comment by flamber** [comment]
Confirmed on stats.

1. Simple question > MongoDB
2. Convert to native query via Editor
3. Save question (works)
4. (refresh browser) Modify the query (add space to the end) and try saving again will change the Save button to "Success", but actual fail, and never close the modal:
`[6e61da27-0b73-44e4-b9b5-30b309089aa2] 2020-04-04T12:50:17+02:00 DEBUG metabase.middleware.log PUT /api/card/1084 400 2.3 ms (0 DB calls) 
{:errors {:result_metadata "value may be nil, or if non-nil, value must be an array of valid results column metadata maps."}}`

Not sure if we need a separate P1 issue for "Success" button even when it fails.

**Comment by flamber** [comment]
Same is happening with Druid (#12285), so this might be Native queries on non-SQL in general.

**Comment by tlrobinson** [comment]
The problem is the `results_metadata` returned for native queries' `POST /api/dataset`  endpoint includes invalid columns. The schema for the columns is:

```clojure
(def ^:private ResultColumnMetadata
  "Result metadata for a single column"
  {:name                          s/Str
   :display_name                  s/Str
   (s/optional-key :description)  (s/maybe su/NonBlankString)
   :base_type                     su/FieldTypeKeywordOrString
   (s/optional-key :special_type) (s/maybe su/FieldTypeKeywordOrString)
   (s/optional-key :unit)         (s/maybe DateTimeUnitKeywordOrString)
   (s/optional-key :fingerprint)  (s/maybe i/Fingerprint)})
```

but for native Mongo (and Druid, I presume) queries it only contains `name` and `fingerprint`, e.x. 

```json
{name: "_id", fingerprint: {global: {distinct-count: 5, nil%: 0}}}
```

The frontend passes the `results_metadata` through directly when saving, which causes the save to fail.

@camsaul @sbelak Can one of you tackle, or delegate this?

Separately, the save modal isn't quite behaving correctly. I filed an issue here: #12291

**Comment by sbelak** [comment]
Seems related to #11256

**Comment by sbelak** [comment]
Same error also interferes with insights on Mongo

**Comment by camsaul** [comment]
Fixed by #12324
